test/test_common/utility.cc

```diff
@@ -173,8 +173,11 @@ AssertionResult TestUtility::waitForCounterEq(Stats::Store& store, const std::st
                                               std::chrono::milliseconds timeout,
                                               Event::Dispatcher* dispatcher) {
   Event::TestTimeSystem::RealTimeBound bound(timeout);
-  while (findCounter(store, name) == nullptr || findCounter(store, name)->value() != value) {
+  while (true) {
     time_system.advanceTimeWait(std::chrono::milliseconds(10));
+    if (findCounter(store, name) != nullptr && findCounter(store, name)->value() == value) {
```

Comment:
Why is this better? The only obvious effect I can see is that in this version, time advances even if, when this is called, the counter is already at the specified value.

Reply:
Actually I faced a problem by the root of previous implementation while working on [this](https://github.com/envoyproxy/envoy/pull/17486). 

The problems I faced will be as follows.

There is a timer that runs at regular intervals, and in the callback of the timer, the counter is incremented. When I tested this process, I ran into a problem. The timer increments the counter while it is waiting by `advanceTimeWait`. And after the wait process, a timeout occurred, resulting in an AssertionFailure.

This PR is to improve this. If there is a better way to prevent this, I will try to adopt it.

Reply:
by these lines of changes https://github.com/envoyproxy/envoy/pull/17486/files#diff-b0bbeea52042998049ac1211b5c7203366ea23f112c0d4f4d939f04697bd0448R84-R120

Reply:
It seems to me like the timeout in your test might not be generous enough, and you are slightly changing the order of operation in this PR rather than adjusting the timeout.

In any case I think there's a better pattern, which is to use a NotifyingAllocator, which provides a much more robust condvar-based mechanism for waiting on counters. See https://github.com/envoyproxy/envoy/blob/cc1d41e7ee9fbfb7ee3c8f73724cdc41d7c6bbb0/test/integration/server.h#L208 .

Is that feasible in your context?

Currently it looks like the notify allocator is tied into the integration test framework. @alyssawilk was there a compelling reason that should only be done in integration tests? Can we use the notifying allocator for unit tests?

Reply:
oh I missed that. let me change existing tests.

Reply:
It worked fine. For now this change is no longer needed I think.

Reply:
sounds good. Closing.